### PR TITLE
chore: add documentation about using import_map with deno_graph

### PR DESCRIPTION
**Pull request by deer**
closes https://github.com/denoland/deno_graph/issues/364

I found the usage of `deno_graph` to be unexpected, and it was a long journey to get where I wanted. Having an example like this (with a focus on using `import_map`) would have saved me a ton of time. I've thoroughly prepared this example with the hope that it helps someone else in the future.

This can be more properly viewed here:
https://github.com/deer/deno_graph/tree/docs_for_import_map/js#exploring-the-entire-dependency-graph

### CODE REVIEW

File: js/README.md [hunk 1]
```diff
@@ -43,3 +43,183 @@ There are several options that can be passed the function in the optional
...
+library provides a `resolve` function which we can pass to `createGraph`.
+
+```ts
+import { createGraph } from "https://deno.land/x/deno_graph@0.63.6/mod.ts";
+import { resolve, toFileUrl } from "https://deno.land/std@0.212.0/path/mod.ts";
+import { parseFromJson } from "https://deno.land/x/import_map@v0.18.3/mod.ts";
+
+const path = resolve("./someFiles/c.ts");
+const importMap =
+  (await import("./deno.json", { with: { type: "json" } })).default;
```
Comment (dsherret) [inline js/README.md:173]: This won't work for jsonc deno.json files. Maybe this should use `Deno.readTextFileSync(new URL("./deno.json", import.meta.url))` instead?

**Review by dsherret — COMMENTED** [review]
Thanks for this PR! It's a little long. Maybe we should just show the code example of how to use it with /x/import_map?

**Comment by deer** [comment]
Thanks for looking. It's much shorter now.

File: js/README.md [hunk 2]
```diff
@@ -43,3 +43,27 @@ There are several options that can be passed the function in the optional
...
+);
+const resolvedImportMap = await parseFromJson(
+  toFileUrl(resolve("deno.json")),
+  importMap,
+);
+const graph = await createGraph(toFileUrl(path).href, {
+  resolve: resolvedImportMap.resolve.bind(resolvedImportMap),
+});
+console.log(graph);
+```
```
Comment (lino-levan) [inline js/README.md:69]: Could you add comments here? The code is relatively clear but I wish it was more clearly explained.

**Review by lino-levan — CHANGES_REQUESTED** [review]

File: js/README.md (hunk 2 above)
Comment (deer) [reply js/README.md]: I added some comments. Let me know what you think.

**Review by dsherret — APPROVED** [review]
Thanks! I updated the docs for the new API on JSR (the import map crate has yet to migrate to JSR, so leaving that url there for now)

**Pull request merged**